cli/src/cli_util.rs

```diff
@@ -1017,15 +1025,29 @@ impl WorkspaceCommandHelper {
                 let mut iter = [commit0, commit1].into_iter().chain(iter);
                 let commits: Vec<_> = iter.by_ref().take(5).try_collect()?;
                 let elided = iter.next().is_some();
-                let hint = format!(
-                    r#"The revset "{revision_str}" resolved to these revisions:{eol}{commits}{ellipsis}"#,
-                    eol = "\n",
-                    commits = commits
-                        .iter()
-                        .map(|c| self.format_commit_summary(c))
-                        .join("\n"),
-                    ellipsis = elided.then_some("\n...").unwrap_or_default()
-                );
+                // Separate hint if the conflict is due to a branch conflict
+                let hint = if let Some(branch_name) = branch {
+                    format!(
+                        r#"The revset "{revision_str}" resolved to these revisions:{eol}{commits}{ellipsis}
+This error is due to a conflicted branch. To resolve this, try setting to which revision the branch points to with "jj branch set {branch_name} --revision <REVISION>""#,
```

Comment:
I think we can make clearer by making it more specialized. Maybe something like this:

```
r#"Branch {branch_name} resolved to multiple revisions because it's conflicted.
It resolved to these revisions:
{commits}{ellipsis}
Use `jj branch set {branch_name} -r` to resolve."#
```